x-pack/plugins/rule_registry/server/utils/rbac.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * registering a new instance of the rule data client
+ * in a new plugin will require updating the below data structure
+ * to include the index name where the alerts as data will be written to.
+ */
+export const mapConsumerToIndexName = {
```

Comment:
It would be great if we could make a mechanism to register these feature ID - index name pairs in plugin setup instead of hardcoding them. Ideally we would integrate that registration with the index template creation and RuleDataClient initialization so when a plugin registers alerts-as-data it automatically sets up RBAC and provides the necessary client(s). The overall goal would be to keep the entire definition for a particular solution's alerts-as-data index inside the plugin that owns it. Right now the index names are specified in multiple places that have to be kept in sync.

Reply:
Def agree. This is an ongoing conversation. As the ruleRegistry right now is a library and meant to be stateless, we didn't have a "home" to store this data we needed in. We're opening up a ticket to discuss where best this all might live.

Reply:
https://github.com/elastic/kibana/issues/104958